lib/testers/rule-tester.js

```diff
@@ -490,7 +490,19 @@ RuleTester.prototype = {
                 }
 
                 if (item.hasOwnProperty("output")) {
-                    const fixResult = SourceCodeFixer.applyFixes(eslint.getSourceCode(), messages);
+                    let fixResult = SourceCodeFixer.applyFixes(eslint.getSourceCode(), messages);
+
+                    if (item.multipass) {
```

Comment:
The current behavior of `RuleTester` is that it only allows you to assert the result of the first pass. This broke five tests (the ones where I added the `multipass` option), since those tests all make two fixes in the same function, which now requires the fixes to be done in two passes.

IMO it seems cleanest to have the test runner do what `cli-engine` does and apply fixes until the code gets to a fixed point, not just apply the first pass, so my first attempt was to change `RuleTester` to always do that. That caused two types of issues:

* A number of tests hard-code the parser output (see `tests/fixtures/parsers`). That parser output is only valid for the first pass, so the second pass would end up using an AST that didn't match the code, which caused all sorts of problems.
* About 20-30 other tests failed because there were further fixes to apply after the first  pass. Most of them were in the tests for the `indent` rule, but there were various others. I tried going through and updating all of them, but decided that that would probably make this change too big.

In the end, I decided to leave all of the other tests alone and just enable this new behavior on the tests that needed it.

When you say this would be a change to core, do you mean that `RuleTester` is a public API that might be used externally? I thought it was just an internal test util and I didn't realize there would be hesitation with changing it.

I could always just get rid of this change and change those five tests to only assert the result of the first pass (which would have one of the fixes but not the other). I was hoping there would still be a way to assert the final result if you actually ran the code through eslint, but I'm certainly fine holding off on that if you are.